Automattic/wp-calypso
# Woop: store onboarding steps data

~### It depends on https://github.com/Automattic/wp-calypso/pull/59785~

#### Changes proposed in this Pull Request

This PR collects and stores the store address into the site options. For this purpose, it implements a new `useSiteOption()` custom hook:

### Only in Development

Check that this `Store Address` step is only visible in development environment.

### Cuctom hook

Ir adds a new custom hook to make the data handling easier.

```es6
import { useSiteOption } from '../hooks/use-site-options';

function StoreAddressFrom() {
  const { get, save, update } = useSiteOptions( 'site-id' );

  return (
    <>
      <TextControl
        label={ __( 'Address line 1' ) }
        value={ get( 'address_1' ) }
        onChange={ ( value ) => update( { address_1: value } ) }
      />

      <Button onClick={ save }>Save</Button>
  </>
  );
}
```

https://user-images.githubusercontent.com/77539/148269203-ded15a22-f38c-4ec5-9d96-99c5b571fc28.mov

#### Testing instructions

* Apply testing patch D72185-code


* Test on development environment
* Fill address data
* Click on `Confirm` button. You can confirm the client performs a POST request in order to save the data
* Confirm the data is persistent, basically via a hard-refresh.
* Confirm next step is `Confirm` step
* Check the back button in the `Confirm` step. It should lead to the `Store Address` step.

Check the feature is not visible on other envs.
You can use the Calypso live branch:
`https://container-flamboyant-banach.calypso.live/woocommerce-installation/<your-testing-site>`

* It should not show the Store Address step.
* Check the back button in the Confirm step. It should lead to the landing page.

<img width="975" alt="Screen Shot 2022-01-06 at 5 26 26 PM" src="https://user-images.githubusercontent.com/1689238/148461100-b28df0ee-eb39-44c9-865b-23997075bc2c.png">



Related to
* https://github.com/Automattic/wp-calypso/issues/57423
* https://github.com/Automattic/wp-calypso/issues/59481
